airbyte-integrations/connectors/source-pinterest/source_pinterest/spec.json

```diff
@@ -0,0 +1,50 @@
+{
+  "documentationUrl": "https://docs.airbyte.io/integrations/sources/pinterest",
+  "connectionSpecification": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Pinterest Spec",
+    "type": "object",
+    "required": ["client_id", "client_secret", "refresh_token", "start_date"],
+    "additionalProperties": false,
+    "properties": {
+      "client_id": {
+        "type": "string",
+        "title": "Client id",
+        "description": "Your Pinterest client id. See the <a href=\"https://developers.pinterest.com/docs/api/v5/#tag/Authentication\">docs</a> for instructions on how to generate it.",
+        "airbyte_secret": true
+      },
+      "client_secret": {
+        "type": "string",
+        "title": "Client secret",
+        "description": "Your Pinterest client secret. See the <a href=\"https://developers.pinterest.com/docs/api/v5/#tag/Authentication\">docs</a> for instructions on how to generate it.",
+        "airbyte_secret": true
+      },
+      "refresh_token": {
+        "type": "string",
+        "title": "Refresh token",
+        "description": "Your Pinterest refresh token. See the <a href=\"https://developers.pinterest.com/docs/api/v5/#tag/Authentication\">docs</a> for instructions on how to generate it.",
+        "airbyte_secret": true
+      },
+      "access_token": {
+        "type": "string",
+        "title": "Access token",
+        "description": "Your Pinterest access token. See the <a href=\"https://developers.pinterest.com/docs/api/v5/#tag/Authentication\">docs</a> for instructions on how to generate it.",
+        "airbyte_secret": true
+      },
+      "start_date": {
+        "type": "string",
+        "title": "Start date",
+        "description": "A date in the format YYYY-MM-DD.",
+        "examples": ["2020-06-01"]
+      },
+      "window_in_days": {
+        "type": "integer",
+        "title": "Window in days",
+        "description": "The amount of days for each data-chunk begining from start_date. Bigger the value - faster the fetch. (Min=1, as for a Day; Max=185, as for a Year).",
+        "examples": [30, 60, 90, 185],
+        "default": 185,
```

Comment:
This option is a little suspicious, I think we should aim to remove it if possible. Otherwise if it has to be included, we need to be very clear about it. The main issue I see with it is: as a user, I don't really understand what the tradeoffs in setting it are. For example, if bigger days are faster, why wouldn't airbyte just use the maximum number always, and make my syncs as fast as possible? My _guess_ (please confirm!) is that it if the user has a lot of data, then setting a large sync window will have a higher probability of the connector failling, because it will run for a longer amount of time at once. 

And even if I know exactly the tradeoff, how does a user know how "much" data they have in their account? Let's say I'm a marketer who only understands how to run campaigns in the UI, how do I know what number to input here? What's "a lot" of data? is it 1 campaign? 10? 500? and what window does it correspond to, 1 day? 30? 185? 

Some more questions: 
1. is it an implementation detail? Why does the user need to care about it? As mentioned in the [UX handbook ](https://docs.airbyte.io/connector-development/ux-handbook#input-configuration), `Avoid configuration at all when possible` and `Avoid exposing implementation details in configuration`
2. Assuming it's important for the user to know about this, what is the tradeoff of inserting different values? if bigger values are faster, why wouldn't the user always select 185? 

Given the principles in the [UX handbook](https://docs.airbyte.io/connector-development/ux-handbook#input-configuration), can you think of a better way to handle this?

Reply:
Agreed with you. Deleted window_in_days from spec param and set it in code to max value (185 days).

Reply:
FWIW I recommend that we set this option to no more than 30 days because otherwise it will cause very long running jobs that don't emit their state very frequently. Feel free to `/publish` after making this change.

Reply:
Updated